app/helpers/details_table_helper.rb

```diff
@@ -1,14 +1,42 @@
 module DetailsTableHelper
-  def details_html(attribute)
+  def details_html(attribute, resource = nil)
+    resource_class = resource.class.name
+
     if attribute[:format] == :bullet && attribute[:value].length > 1
       list = attribute[:value].map { |value| "<li>#{value}</li>" }.join
       simple_format(list, { class: "govuk-list govuk-list--bullet" }, wrapper_tag: "ul")
     else
-      return simple_format(attribute[:value].first.to_s, { class: "govuk-body" }, wrapper_tag: "p") if attribute[:value].is_a?(Array)
-
-      value = attribute[:value].presence || "<span class=\"app-!-colour-muted\">You didn’t answer this question</span>".html_safe
+      return simple_format(attribute[:value].first.to_s, { class: "govuk-body" }, wrapper_tag: "p") if attribute[:value].is_a?(Array) && attribute[:value].any?
 
+      value = determine_value(attribute, resource, resource_class)
       simple_format(value.to_s, { class: "govuk-body" }, wrapper_tag: "p")
     end
   end
+
+private
+
+  def determine_value(attribute, resource, resource_class)
+    attribute[:value].presence || case resource_class
+                                  when "Location"
+                                    location_value(attribute, resource)
+                                  when "Organisation"
+                                    organisation_value(attribute, resource)
+                                  when "Scheme"
+                                    scheme_value(attribute, resource)
+                                  else
+                                    "<span class=\"app-!-colour-muted\">No answer provided</span>".html_safe
+                                  end
+  end
```

Comment:
`attribute[:value].presence || send("#{resource_class.downcase}_value", attribute, resource) || "<span class=\"app-!-colour-muted\">No answer provided</span>".html_safe`

or early returns

Reply:
Nice i'll try that. I can try remove `No answer provided` from all places regarding logs but the other pages (e.g. Users, Schemes...) might need to remain separate.